Source/Scene/PolygonClippingAccelerationGrid.js

```diff
@@ -0,0 +1,505 @@
+
+  var row;
+  var col;
+
+  for (row = 0; row < this.numRows; ++row) {
+    for (col = 0; col < this.numCols; ++col) {
+      var cellTopLeft = {
+        x: bbox.topLeft.x + col * this.cellWidth,
+        y: bbox.topRight.y - this.cellHeight * row,
+      };
+
+      var cellTopRight = {
+        x: bbox.topLeft.x + (col + 1) * this.cellWidth,
+        y: cellTopLeft.y,
+      };
+
+      var cellBtmRight = {
+        x: cellTopRight.x,
+        y: bbox.topRight.y - this.cellHeight * (row + 1),
+      };
+
+      var cellBtmLeft = {
+        x: cellTopLeft.x,
+        y: cellBtmRight.y,
+      };
+
+      var cell = {
+        topLeft: cellTopLeft,
+        topRight: cellTopRight,
+        btmRight: cellBtmRight,
+        btmLeft: cellBtmLeft,
+      };
+
+      var cellIsIndeterminate = false;
+      var cellIsCompletelyOccluded = false;
+      var partiallyOverlappingTriangleIndices = [];
+
+      for (var i = 0; i < indices.length; i += 3) {
+        var ax = positions[indices[i] * 3];
+        var ay = positions[indices[i] * 3 + 1];
+        var bx = positions[indices[i + 1] * 3];
+        var by = positions[indices[i + 1] * 3 + 1];
+        var cx = positions[indices[i + 2] * 3];
+        var cy = positions[indices[i + 2] * 3 + 1];
+
+        // if the edge of this triangle intersects with the cell
+        // then the cell is indeterminate
+        if (doesTriangleIntersectRect(cell, ax, ay, bx, by, cx, cy)) {
+          cellIsIndeterminate = true;
+          var i0 = indices[i];
+          var i1 = indices[i + 1];
+          var i2 = indices[i + 2];
+          partiallyOverlappingTriangleIndices.push(i0, i1, i2);
+        }
+
+        if (doesTriangleOverlapRect(cell, ax, ay, bx, by, cx, cy)) {
+          cellIsCompletelyOccluded = true;
+          cellIsIndeterminate = false;
+          partiallyOverlappingTriangleIndices = [];
+          break;
+        }
+      }
+
+      var cellIndex = (row * this.numCols + col) * cellNumElements;
+
+      // if the cell is indeterminate we record all the overlapping
+      if (cellIsIndeterminate) {
+        this.grid[cellIndex] =
+          PolygonClippingAccelerationGrid.CellOcclusion.Partial;
+        var startIndex = allOverlappingTriangles.length;
+        var endIndex = startIndex + partiallyOverlappingTriangleIndices.length;
+        this.grid[cellIndex + 1] = startIndex;
+        this.grid[cellIndex + 2] = endIndex;
+        allOverlappingTriangles = allOverlappingTriangles.concat(
+          partiallyOverlappingTriangleIndices
+        );
+      } else if (cellIsCompletelyOccluded) {
+        this.grid[cellIndex] =
+          PolygonClippingAccelerationGrid.CellOcclusion.Total;
+      } else {
+        this.grid[cellIndex] =
+          PolygonClippingAccelerationGrid.CellOcclusion.None;
+      }
+    }
+  }
+
+  this.overlappingTriangleIndices = Float32Array.from(allOverlappingTriangles);
+}
+
+PolygonClippingAccelerationGrid.prototype.getCellFromWorldPosition = function (
```

Comment:
Does this only exist for testing purposes? If so, just mark is as such. Usually we do that by adding an underscore to the function name and add a comment. `PolygonClippingAccelerationGrid.prototype._getCellFromWorldPosition`